service/history/replication/task_processor.go

```diff
@@ -477,15 +477,16 @@ func (p *taskProcessorImpl) processTaskOnce(replicationTask *types.ReplicationTa
 			}
 			mScope = mScope.Tagged(metrics.DomainTag(domainName))
 		}
-		// emit the number of replication tasks
-		mScope.IncCounter(metrics.ReplicationTasksAppliedPerDomain)
 		// emit single task processing latency
 		mScope.RecordTimer(metrics.TaskProcessingLatency, now.Sub(startTime))
 		// emit latency from task generated to task received
 		mScope.RecordTimer(
 			metrics.ReplicationTaskLatency,
 			now.Sub(time.Unix(0, replicationTask.GetCreationTime())),
 		)
+		// emit the number of replication tasks
+		shardScope := mScope.Tagged(metrics.InstanceTag(strconv.Itoa(p.shard.GetShardID())))
```

Comment:
I think you need to rescope here. Otherwise only PerDomain metrics would have shard metrics, because root has not shard. 

On second thought, this would actually blow the cardinality with per domain + per shard. What we actually need here is per shard while keeping per domain. It means we might need to create a separate metrics `ReplicationTasksAppliedPerShard`